nailgun-examples/src/main/java/com/martiansoftware/nailgun/examples/Heartbeat.java

```diff
@@ -42,21 +39,14 @@ public static void nailMain(final NGContext context) throws IOException {
         try {
             // Register a new NGClientListener. As clientDisconnected is called from
             // another thread any nail state access must be properly synchronized.
-            context.addClientListener(new NGClientListener() {
-                public void clientDisconnected() throws InterruptedException {
-                    throw new InterruptedException("Client Disconnected"); // Will interrupt the thread below.
-                }
-            });
+            Thread mainThread = Thread.currentThread();
+            context.addClientListener(() -> mainThread.interrupt());
```

Comment:
nit: you should be able to use a method reference here. Something like `addClientListener(mainThread::interrupt)`